engine/src/flutter/impeller/display_list/canvas.cc

```diff
@@ -1912,32 +1914,28 @@ std::shared_ptr<Texture> Canvas::FlipBackdrop(Point global_pass_position,
   // In cases where there are no contents, we
   // could instead check the clear color and initialize a 1x2 CPU texture
   // instead of ending the pass.
-  rendering_config.inline_pass_context->GetRenderPass();
-  if (!rendering_config.inline_pass_context->EndPass()) {
+  rendering_config.GetInlinePassContext()->GetRenderPass();
+  if (!rendering_config.GetInlinePassContext()->EndPass()) {
     VALIDATION_LOG
         << "Failed to end the current render pass in order to read from "
            "the backdrop texture and apply an advanced blend or backdrop "
            "filter.";
     // Note: adding this render pass ensures there are no later crashes from
     // unbalanced save layers. Ideally, this method would return false and the
     // renderer could handle that by terminating dispatch.
-    render_passes_.push_back(LazyRenderingConfig(
-        renderer_, std::move(rendering_config.entity_pass_target),
-        std::move(rendering_config.inline_pass_context)));
+    render_passes_.emplace_back(std::move(rendering_config));
```

Comment:
It feels like a bunch of places put the same config back into the passes (to avoid a crash) except one that seems to be deemed "successful" which replaces it with a different config.

Wouldn't it be simpler to just leave the pass in the vector until we know it is something we can act on and only swap it out in that one case?

That would avoid the potential for someone adding another failure check and not balancing the passes.

Reply:
Also not a bad idea, but outside of the scope of the change I was making.  I'd prefer to leave it untouched.